src/components/Toggle/index.js

```diff
@@ -0,0 +1,32 @@
+import './index.scss';
+import { Component, createIDFactory } from 'pet-dex-utilities';
+
+const generateID = createIDFactory('toggle');
+
+const events = ['toggle'];
+
+const html = `
+  <div class="toggle-container">
+    <input type="checkbox" class="toggle-container__input" data-select="toggle-input">
+    <label class="toggle-container__label" data-select="toggle-label"></label>
+  </div>
+`;
+
+export default function Toggle({ checked = false } = {}) {
+  Component.call(this, { html, events });
+
+  const id = generateID();
+  this.selected.get('toggle-input').setAttribute('id', id);
+  this.selected.get('toggle-label').setAttribute('for', id);
+
+  this.selected.get('toggle-input').checked = checked;
+
+  this.selected.get('toggle-input').addEventListener('change', () => {
+    this.emit('toggle', this.selected.get('toggle-input').checked);
+  });
+}
+
+Toggle.prototype = Object.assign(
+  Toggle.prototype,
+  Component.prototype,
+);
```

Comment:
Aqui precisamos de um método chamado `setToggle` cujo recebe um booleano e altera o estado programaticamente do toggle.

Essa alteração provavelmente já vai ativar o emit do toggle, então **não precisa** emitir o evento!

(provavelmente o método que o diogo ta falando é esse que to pedindo)